week3/src/components/Content.jsx

```diff
@@ -0,0 +1,219 @@
+                }
+            }
+        `
+    }
+    ${props => props.score == 5 &&
+        css`
+            background-color: #89c8ff;
+            animation-name: pop5;
+            animation-duration: .2s;
+
+            @keyframes pop5 {
+                0% {
+                transform: translate(0, 0);
+                }
+                50% {
+                transform: translate(0, -30px);
+                }
+                100% {
+                transform: translate(0, 0);
+                }
+            }
+        `
+    }
+
+    font-size: 20px;
+    font-weight: 700;
+
+    border-radius: 30px;
+`;
+
+const TestArticle = styled.article`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+`
+const EndArticle = styled.article`
+    display: flex;
+    justify-content: center;
+    flex-direction: column;
+
+    height: 300px;
+`;
+
+const Image = styled.img`
+    width: 300px;
+
+    margin: 20px;
+
+    border-radius: 20px;
+`
+
+const Buttons = styled.div`
+    margin: 10px;
+`
+
+const Button = styled.button`
+    margin: 10px;
+    padding: 10px 30px;
+
+    border: 0;
+    border-radius: 20px;
+
+    cursor: pointer;
+`;
+
+const members = [
+    "남준", "림", "명지", "명헌", "서연", "서현", "서영", "서현",
+    "서희", "성경", "예원", "우영", "유진", "은형", "재욱", "주함",
+    "준상", "지수", "지영", "하윤", "현수", "현욱", "현지", "형겸", "혜은"
+];
+
+export default function Content({modalOpen, setModalMessage}) {
+    const [score, setScore] = useState(0);
+    // 프로필 셔플 -> 옵션 -> 옵션 중 랜덤하나로 answer
+    const [profiles, setProfiles] = useState(members.map((member) => (
+        {
+            name: member,
+            image : `images/${member}.jpg`
+        }
+    )));
+    const [options, setOptions] = useState(profiles.slice(0, 5));
+    const [answer, setAnswer] = useState(options[parseInt(Math.random() * 5)]);
+
+    useEffect(() => {
+        setAnswer(options[parseInt(Math.random() * 5)]);
+    }, [options]);
+
+    useEffect(() => {
+        setOptions(profiles.slice(0, 5))
+    }, [profiles]);
+
+    const onClickOption = (e) => {
+        if(e.currentTarget.innerText === answer.name) {
+            // 정답일 경우
+            setScore((prev) => prev + 1);
+            setProfiles((prev) => [...prev].sort(() => Math.random() - 0.8));
+            setModalMessage("정답!");
+        } else {
+            // 땡
+            setModalMessage("틀렸어... 실망이야.");
+        }
+        modalOpen();
+    }
+
+    const onClickRestart = () => {
+        window.location.reload();
+    }
+    
+  return (
+    <Section>
+        <Score score={score}>{score}</Score>
+        {score >= 5 ?
+        <EndArticle>
+            <h2>🥳🥳🥳🥳🥳</h2>
+            <h2>내가 해냄~~</h2>
+        </EndArticle>
+        : <TestArticle>
+            <Image src={answer.image}/>
+            <Buttons>
+                {options.map((option) => (
+                    <Button onClick={onClickOption}>{option.name}</Button>
+                ))}
```

Comment:
그냥 이렇게 사용함으로써 option state 를 사용하지 않아도 되지 않을까요??

```suggestion
                {profiles.slice(0, 5).map((option) => (
                    <Button onClick={onClickOption}>{option.name}</Button>
                ))}
```

Reply:
사실 이거 PR 쓰다보니 고쳐야겠다고 생각했음... state를 최대한 줄이는게 좋겠죠?!